Cargo.toml

```diff
@@ -28,6 +28,9 @@ tempfile = { version = "3.5.0", optional = true }
 anyhow = { version = "1.0.34", optional = true }
 clap = { version = "4.1", features = ["derive", "cargo"], optional = true }
 
+serde = "1.0"
+serde_json = "1.0"
```

Comment:
Can we make these optional dependencies?

Reply:
I imagine the benefit is that when this is used as a library, downstream does not need to support these dependencies?  Yes, certainly.  I'll get that done.

Reply:
Added a feature flag that enables serde.  Would you like serde to be enabled by default or not?  I have pushed a commit in which it is _not_ enabled by default, so that existing downstream users of the library are unaffected by the change but I can enable it by default, if you wish.